### ISSUE TITLE
[1304] Slide Up Overlay

### ISSUE THREAD

**Issue by athomasmoz** [mozilla-mobile/firefox-tv#1310]
### Why/User Benefit/User Problem

As a user, I get confused about where I am in the app. If I'm on a web page and hit the hamburger button, I seem to be taken back to the app's homescreen, but then if I hit the back button, I return to the website I was viewing.

We also want to improve the separations of concerns between website navigation actions and content actions. We also assume that splitting the toolbar from the rest of the content will help users understand how the their inputs will affect the website that they can still see underneath.

### What / Requirements

- Slide up the overlay when the hamburger button is pressed from a web page
- Keep the browser toolbar at the top of the page
- ~@aminalhazwani can you add any more details on the behaviour here?~

![image](https://user-images.githubusercontent.com/43346208/47120789-d80a2800-d224-11e8-81df-cefabba44313.png)

This is part of epic #1304

**Comment by aminalhazwani** [comment]
While browsing on press of the MENU button the toolbar will slide in from the top and the channels will slide in from the bottom. The focus will be on the location bar.

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/1721875/47561415-83009d00-d91b-11e8-8da4-44a2e6371870.png">

Moving DOWN with the d-pad will focus the immediate channel and align said channel to half the height of the screen.

<img width="995" alt="image" src="https://user-images.githubusercontent.com/1721875/47570307-26f74200-d936-11e8-8249-6ade6944fbfa.png">

To close toolbar and channels users need to press BACK button on the remote. This is necessary since we use the MENU button for contextual features (see remote hints). If they try to go back to browsing via the MENU button we could display a notification that hints them to press the BACK button.

<img width="998" alt="image" src="https://user-images.githubusercontent.com/1721875/47572248-d209fa80-d93a-11e8-9517-2515d41927dd.png">

**Comment by athomasmoz** [comment]
Waiting on #1394

**Comment by Baron-Severin** [comment]
Unassigning myself because I'll be working on Echo Show by the time this is unblocked.

**Comment by liuche** [comment]
Also, note on possibly laggy perf seen in FFES, due to holding onto all fragment state: https://github.com/mozilla-mobile/firefox-tv/pull/1637#discussion_r245449221

**Comment by athomasmoz** [comment]
Closing in favour of splitting into separate tickets.
- Refactor overlay into split #1664 
- Animate overlay entrance/removal #1666
- Center selected channel on screen #1312
